### ISSUE TITLE
Mesh created from SCNNode and vice versa

### ISSUE THREAD

**Issue by tonyskansf** [nicklockwood/Euclid#86]
Hi 👋 

It seems that each created `Mesh` from `SCNNode` is positioned at the origin (zero vector) regardless of the node's `position` property in the scene. Am I creating meshes the wrong way, or is this an expected behavior?

Example:

```swift
let cubeMesh = Mesh.cube(size: 0.2)
let cubeGeometry = SCNGeometry(cubeMesh)
var cubeNode = SCNNode(geometry: cubeGeometry)

// Simulate position changes
cubeNode.position = SCNVector3(0, 0.2, 0)

// Create a mesh from the node (at the desired position)
let mesh = Mesh(cubeNode)
    //.translated(by: Vector(cubeNode.position)) // This solves the problem and the mesh represents the node at the same position.

// Display the created Mesh
// Will appear at zero vector and not (0, 0.2, 0)
let nodeFromMesh = SCNNode(geometry: SCNGeometry(mesh)) 
```

My use case is the subtract operation on two nodes sometime later after their initialization. However, doing this with the meshes created the way shown above does not work because they are both at the origin. 

The solution here is to translate both meshes by the node's position and then do the operation. This works; however, only if both nodes are discarded, and the resulting mesh is displayed as a new node. In my case, I need to replace the node's geometry—the one calling the subtract method—with the new one representing the subtracted state. But this shows the node at the wrong position as if the translation was applied.

```swift
let mesh = Mesh(cubeNode).translated(by: Vector(cubeNode.position))
let meshToSubtract = Mesh(otherNode).translated(by: Vector(otherNode.position))
let result = mesh.subtract(meshToSubtract)

cubeNode.geometry = SCNGeometry(result) // Shows the node moved by the position
let subtractedNode = SCNNode(geometry: SCNGeometry(result)) // Appears at the correct position
```

So far I've resolved this by translating the resulting mesh back to the correct position but that just feels wrong.
```swift
let position = -1 * cubeNode.position
let result = mesh
    .subtract(meshToSubtract)
    .translated(by: Vector(position))
```

Do you see anything that is entirely wrong here? Any help is much appreciated! 🙏

**Comment by nicklockwood** [comment]
This was intentional, but I can see why it would seem like a bug. I'll consider changing it in future (although I'll need to consider backwards compatibility). In the meantime your solution seems fine.

**Comment by tonyskansf** [comment]
Nice, I believe it would be a great addition to the framework. Thank you for the answer.

**Comment by nicklockwood** [comment]
@tonyskansf this is fixed in 0.5.18. I've added an explicit parameter to control whether the transform is applied.
